components/VM_Arm/src/main.c

```diff
@@ -815,8 +847,6 @@ static int load_linux(vm_t *vm, const char *kernel_name, const char *dtb_name, c
     }
 
     if (!config_set(CONFIG_VM_DTB_FILE)) {
-        camkes_io_fdt(&(_io_ops.io_fdt));
```

Comment:
I've seen similar changes to this in a patch by @WellMcGarnicle [1].  In that patch, instead of adding a new config option, the patch assumes that if a dtb_name attribute is set in the camkes config it refers to a file in the file server (as that's what the variable's purpose is). So if CONFIG_VM_DTB_FILE is not set, it tries to get the file from the file server but falls back to the camkes_io_fdt if the file doesn't exist. 

[1]
```
diff --git a/components/VM_Arm/src/main.c b/components/VM_Arm/src/main.c
index 2b14989..07ab026 100644
--- a/components/VM_Arm/src/main.c
+++ b/components/VM_Arm/src/main.c
@@ -104,6 +104,7 @@ vmm_io_port_list_t *io_ports;
 reboot_hooks_list_t reboot_hooks_list;
 #define PLAT_LINUX_DTB_SIZE 0x50000
 static char linux_gen_dtb_buf[PLAT_LINUX_DTB_SIZE];
+static char linux_original_dtb_buf[PLAT_LINUX_DTB_SIZE];
 
 struct ps_io_ops _io_ops;
 
@@ -829,8 +830,23 @@ static int load_linux(vm_t *vm, const char *kernel_name, const char *dtb_name, c
     }
 
     if (!config_set(CONFIG_VM_DTB_FILE)) {
-        camkes_io_fdt(&(_io_ops.io_fdt));
-        void *fdt_ori = ps_io_fdt_get(&_io_ops.io_fdt);
+        void *fdt_ori = NULL;
+        // Check if we've been given a dtb to use as a base for generation.
+        int dtb_fd = open(dtb_name, 0);
+        if (dtb_fd == -1) {
+          printf("FDT Generation: Using camkes io dtb as base\n");
+          // No DTB found, using the camkes built in dtb.
+          camkes_io_fdt(&(_io_ops.io_fdt));
+          fdt_ori = ps_io_fdt_get(&_io_ops.io_fdt);
+        } else {
+          printf("FDT Generation: Using provided dtb ('%s') as base\n", dtb_name);
+          size_t len = read(dtb_fd, linux_original_dtb_buf, PLAT_LINUX_DTB_SIZE);
+          if (len <= 0) {
+            ZF_LOGE("Failed to read provided dtb");
+            return -1;
+          }
+          fdt_ori = (void *)linux_original_dtb_buf;
+        }
 
         void *gen_fdt = linux_gen_dtb_buf;
         int size_gen = PLAT_LINUX_DTB_SIZE;
```

Reply:
Seems really close to what I'm trying to achieve here. Do you want to pull that in, or should I update my PR to make the changes more similar to that?

Reply:
Yea, for a second I thought they were the same implementation initially. It's fine to merge this one if you're also happy removing the config option.